frappe/public/js/frappe/list/list_view.js

```diff
@@ -407,11 +403,12 @@ frappe.views.ListView = class ListView extends frappe.views.BaseList {
 
 		this.columns = this.columns.slice(0, this.list_view_settings.total_fields || total_fields);
 
-		if (
-			!this.settings.hide_name_column &&
-			this.meta.title_field &&
-			this.meta.title_field !== "name"
```

Comment:
Why removed these two conditions?

Reply:
`title_field` can't be set to `"name"`, so the second condition is always true, but I see that on the code level it might:

```python
	def get_title_field(self):
		"""Return the title field of this doctype,
		explict via `title_field`, or `title` or `name`"""
		title_field = getattr(self, "title_field", None)
		if not title_field and self.has_field("title"):
			title_field = "title"
		if not title_field:
			title_field = "name"
```

Hence this was a mistake.

Reply:
Fixed in 8c4d3e5ed9